workers/task-driver/src/driver.rs

```diff
@@ -0,0 +1,298 @@
+//! The task driver drives a task forwards and executes partial retries
```

Comment:
mostly moved over, the implementation of `start_task` has changed and the `run` method is new